### PR TITLE
Ramya teams limit see all

### PR DESCRIPTION
**Pull request by scoopstroop92**
# Description of the feature

- A “Limit See-All” option for teams 

-  Other Links ->Teams

- When this option is turned on for a team member,  the team member  shouldn't  be able to see  other members of the team in their dashboard 

- The member however should be able to see team members of other teams  they are part of 

## Related PRS (if any):
This frontend PR is related to the [576](https://github.com/OneCommunityGlobal/HGNRest/pull/576) backend PR.

## Main changes explained:
-  Added a column ‘See all’ in the "Team Menber " popup
-  A tool tip beside See All
-  Added a Toggle Switch in Add/Edit Teams pop-up beside every row 
-  Renderd the toggle switch with apprpriate choice.
-  When the user makes the choice,  make a call to   action function 
-  The  action fiunction  makes a post call to the backend 
-  Make a dispatch call to the reducer to update the redux store. 

## How to test:
1.  check into current branch
2.  do `npm install` and `...` to run this PR locally
3.  log as admin user
4.  Go to Other Links ->Teams
5.  Create two Teams , lets say Team A and Team B (you can use any existing Team as well)
6.  Add  the same  team member to Team A and Team B
7.  Enable the toggle switch in Team A for this member
8.  Login to the user's profile and verify if the user is only able  to see  the team members of the  team B and not Team A

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/89441435/2eaaf5a2-9f80-442c-9ace-20fddc997f35

## Note:

- Kindly mention any  bugs , errors you see in the console. 
- Try breaking the functionality , any suggestion is helpful

### CODE REVIEW

**Review by DavidC0126 — DISMISSED** [review]
Hi @ramyaram2092 I tested this PR with BE [PR#576](https://github.com/OneCommunityGlobal/HGNRest/pull/576) and it works as intended! Good job!


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/74082492/86a87d53-874c-4eca-9f46-64ee0e3cf303

**Review by shubhankarval — CHANGES_REQUESTED** [review]
Hey @ramyaram2092, I would like to request 3 changes. 

1. Please change the text in the popup, its kind of confusing. Does enabling the option mean 'allowing the user to see all team members'?
pr

2.  The state of the toggle is not preserved. 
After enabling the toggle:
pr1
After closing and reopening the members popup:    
pr2

3. If possible, can you make the toggle open to the right and not to the left, as that is the commonly used pattern. This would mean the toggle would be disabled (gray) on the left side, and enabled (blue) on the right side.

**Review by StrawberryCalpico — CHANGES_REQUESTED** [review]
The functionality to let user unsee their team members work, great job! but the state of the toggle will be reset if i close the pop up window. Since the toggle is reset every time, there is currently no way to make the user able to see their team members again.


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/114443664/0334f792-6141-4738-8641-90d221fb645e

**Review by ChirayuRai — CHANGES_REQUESTED** [review]
Hey! It seems that setting the toggle sets the value to false, and I think that doesn't preserve the state for being able to view other members. I think that's also why that option doesn't persist when closing the popup.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/15188751/b3be4ae0-2679-46b6-8b03-95c1e5f2e92a

**Comment by ramyaram2092** [comment]
I am investigating the bug/ feedbacks received so far. Marking the PR to "do not review " for now on the same account

**Comment by ramyaram2092** [comment]
Hello @ChirayuRai  @shubhankarval @StrawberryCalpico I have addressed your feedbacks. Could you  re-test and let me know your observation?

**Review by Alforoan — DISMISSED** [review]
It works as intended. Nice!


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/118868270/79e3c2f8-1b69-4a2b-aa87-4a2b4d4c273e

**Comment by keyunhuangg** [comment]
I added the same members of both team a and team b. And disable the button for in team a for keyun(testing) huang. the teams in profiles do not show team a. and there is no member in the leaderboard.

Screenshot 2023-12-23 at 10 17 39 AM
Screenshot 2023-12-23 at 10 20 18 AM
Screenshot 2023-12-23 at 10 17 44 AM
Screenshot 2023-12-23 at 10 24 18 AM

**Review by sucheta90 — COMMENTED** [review]
Hi @ramyaram2092 , I've successfully conducted testing on this PR. I logged in as an Administrator and created two teams containing identical members. Subsequently, I disabled visibility for one team member. Upon logging in as the specified user, the expected outcome was observed: I couldn't view the team members.
Screenshot 2023-12-23 at 6 33 44 PM
Screenshot 2023-12-23 at 6 32 56 PM

**Review by ljrirene — CHANGES_REQUESTED** [review]
Hi Ramya, I tested your PR and I found something which is unexpected. 
1. The toggle switch is not preserved, like other people mentioned under this PR. It is hard to make sure if it is set successfully.
2. I logged in as a manager user, but when I checked the team members, I can still see the volunteer user and they are under team a and team b. Could you please explain more  about how to test this PR?
Thank you! Here is my test video:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/34266880/8d096cff-f01f-4a56-a555-817c194824f9

**Comment by mianmiantea2019** [comment]
Hi Ramya, 

I tested your PR, here is what I've found.

1. Same as like other people mentioned under this PR. The toggle switch is not preserved.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/59427068/c306b4f8-ec39-4726-b291-f3688c7cf03e


2. I  can see the all team members on the dashboard.

image

**Review by X1aoZhang99 — COMMENTED** [review]
Hi Ramya,

I've reviewed your PR and the functionalities work as intended. 

The toggle switch status is now preserved.  

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/113070229/0e783bd6-bdf7-4d6e-9f5f-6732bedb8ac0

**Review by X1aoZhang99 — APPROVED** [review]

**Comment by yuhuimin1996** [comment]
Hi Ramya,
I've tested this FE pr along with the BE pr#576. I could toggle the switcher of my test role under the admin account, while I didn't find where is the team member will show on the dashboard or in the profile. Please see more details in the video:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108281570/9a3f7451-5eda-4cef-8780-d19dffb52b24

**Review by pshereen — COMMENTED** [review]
Tasks are not loading in the dashboard after I check into the "ramya-teams-limit-see-all" branch.
Everything is working fine in the development branch.

File: src/components/UserProfile/UserProfileEdit/ToggleSwitch/ToggleSwitch.jsx [hunk 1]
```diff
@@ -1,8 +1,9 @@
-import React from 'react';
+import { useState } from 'react';
 import style from './ToggleSwitch.module.scss';
 import TriStateToggleSwitch from './TriStateToggleSwitch';
 
-const ToggleSwitch = ({ switchType, state, handleUserProfile, fontSize }) => {
+const ToggleSwitch = ({ switchType, state, handleUserProfile, fontSize, UpdateTeamMembersVisiblity, userId, choice }) => {
+  const[visiblity,setVisiblity]=useState(choice);
```
Comment (rusalshrestha98) [inline src/components/UserProfile/UserProfileEdit/ToggleSwitch/ToggleSwitch.jsx:6]: Nitpick but the spacing on this line could probably be improved.
`const [visiblity, setVisiblity] = useState(choice);`

File: src/components/Teams/TeamMembersPopup.jsx [hunk 2]
```diff
@@ -1,27 +1,44 @@
...
-import { Button, Modal, ModalHeader, ModalBody, ModalFooter, Container, Alert } from 'reactstrap';
+import {
+  Button,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  Container,
+  Alert,
+  TabContent,
```
Comment (rusalshrestha98) [inline src/components/Teams/TeamMembersPopup.jsx:10]: Don't see `TabContent` being used anywhere in this component, should it be removed?

File: src/components/Teams/TeamMembersPopup.jsx [hunk 3]
```diff
@@ -1,27 +1,44 @@
...
 import { faSort, faSortUp, faSortDown } from '@fortawesome/free-solid-svg-icons';
 import { connect } from 'react-redux';
+import ToggleSwitch from '../UserProfile/UserProfileEdit/ToggleSwitch';
+import InfoModal from './InfoModal';
 
 export const TeamMembersPopup = React.memo(props => {
   const closePopup = () => {
     props.onClose();
-    setSortOrder(0)
+    setSortOrder(0);
```
Comment (rusalshrestha98) [inline src/components/Teams/TeamMembersPopup.jsx:27]: `setSortOrder(0)` is being used before it is defined so I would recommend moving the `closePopup()` function below line 35 after the declared states. 

You typically want to declared all the variables and then the functions of the component.

File: src/components/Teams/TeamMembersPopup.jsx [hunk 4]
```diff
@@ -103,29 +122,40 @@ export const TeamMembersPopup = React.memo(props => {
...
+    setTeamVisibility(props.teamData);
+  }, [props.members.teamMembers, sortOrder]);
 
   useEffect(() => {
     onValidation(true);
   }, [props.open]);
 
+  useEffect(() => {
+    setTeamVisibility(props.teamData);
+  }, [props.teamData]);
```
Comment (rusalshrestha98) [inline src/components/Teams/TeamMembersPopup.jsx:153]: Nitpick: add a line break after this `useEffect()` to improve code readability and consistency of spacing.

**Review by rusalshrestha98 — CHANGES_REQUESTED** [review]
Please see comments above.

**Review by Renanluizssx — COMMENTED** [review]
Hi @ramyaram2092 , i don't know if i did something wrong, but when i add a team member on the team a and b, the team members didn't save, you can look my review below:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/91035340/f4139ec8-d12a-428d-bf75-17d378d5e88f

**Comment by ramyaram2092** [comment]
Hi all thanks for your detailed feedback . I am looking into your feedback and  addressing them, marking the PR  do not review on the same note

**Comment by ramyaram2092** [comment]
Addressed all the issue in [1905] (https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/1905). Closing the PR on the same note

**Pull request closed without merging**